src/mqt/predictor/rl/PredictorEnv.py

```diff
@@ -251,8 +267,8 @@ def determine_valid_actions_for_state(self) -> list[int]:
         if mapped and self.layout is not None:
             return [self.action_terminate_index, *self.actions_opt_indices]  # type: ignore[unreachable]
 
-        if self.state._layout is not None:  # noqa: SLF001
-            return self.actions_routing_indices
+        if self.layout is not None:
+            return self.actions_routing_indices  # type: ignore[unreachable]
```

Comment:
Is this really just a false positive from mypy or is something not quite right here?
This warning would indicate that `self.layout` is always `None` here.

Reply:
When initializing the `PredictorEnv`, the attribute is set to `None` and is only set by a `Layout` action. This is what leads to mypy's assumption.

Reply:
Still kind of strange, but ok.